### PR TITLE
[Feature] Add automatic claim of bonus points

### LINKED ISSUE #3616: Automatically claim bonus channel points
Heya,

So one of my favourite streamers just got channel points, and I noticed the bonus button that pops up from time to time. I think a good feature would be to automatically claim these points for the user so they don't miss out if, for example, they go AFK. Would love to see it!

Thanks,
Elliott

### PR DESCRIPTION
**Pull request by whitecrownclown**
Closes #3616

@night Don't think this deserves a setting, what do you think?

Thanks!

### CODE REVIEW

**Comment by night** [comment]
thanks for the PR. imo this definitely needs a setting, since it performs an action on behalf of the user

**Comment by whitecrownclown** [comment]
I've added a setting for it.

**Comment by pmrt** [comment]
Any news on this one @night? Looking forward to this feature!

**Comment by whitecrownclown** [comment]
@night Hey, any updates on how we can get this merged? Thanks!

File: src/modules/channel_points/index.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    })
+);
+
+class ChannelPoints {
+    constructor() {
+        settings.add({
+            id: 'autoClaimBonusPoints',
+            name: 'Automatic Claim of Bonus Points',
+            defaultValue: false,
+            description: 'Automatically claim channel bonus points'
```
Comment (night) [inline src/modules/channel_points/index.js:22]: let's reword this to "Auto-claim Channel Points" and "Automatically claims bonus channel points"

File: src/modules/channel_points/index.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+    autoClaimBonusPoints() {
+        if (!settings.get('autoClaimBonusPoints')) {
+            bonusPointsObserver.disconnect();
+            return;
+        }
+
+        $(claimButtonSelector).click();
+
+        const observe = (_watcher, element) => {
```
Comment (night) [inline src/modules/channel_points/index.js:46]: move this to the new dom observer

Comment (whitecrownclown) [reply src/modules/channel_points/index.js:46]: done!

File: src/modules/channel_points/index.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+            channelPointsListener = domObserver.on(
+                CLAIM_BUTTON_SELECTOR,
+                (node, isConnected) => {
+                    if (!isConnected) return;
+                    const $node = $(node);
+
+                    $node.click();
+                },
+                { useParentNode: true }
```
Comment (night) [inline src/modules/channel_points/index.js:44]: im surprised the linter is not mad. can you remove these spaces

File: src/modules/channel_points/index.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+        });
+
+        settings.on('changed.hideChannelPoints', this.hideChannelPoints);
+        settings.on('changed.autoClaimBonusPoints', this.autoClaimBonusPoints);
+
+        this.autoClaimBonusPoints();
+        watcher.on('load.channel', this.autoClaimBonusPoints);
+    }
+
+    autoClaimBonusPoints() {
```
Comment (night) [inline src/modules/channel_points/index.js:32]: nit: `toggleAutoClaimBonusPoints`

File: src/modules/channel_points/index.js [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+            return;
+        }
+
+        if (!channelPointsListener) return;
+
+        channelPointsListener();
+        channelPointsListener = undefined;
+    }
+
+    hideChannelPoints() {
```
Comment (night) [inline src/modules/channel_points/index.js:56]: nit: `toggleHideChannelPoints`

File: src/modules/channel_points/index.js [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+            name: 'Hide Channel Points',
+            defaultValue: false,
+            description: 'Hides channel points from the chat UI to reduce clutter',
+        });
+
+        settings.on('changed.hideChannelPoints', this.hideChannelPoints);
+        settings.on('changed.autoClaimBonusPoints', this.autoClaimBonusPoints);
+
+        this.autoClaimBonusPoints();
+        watcher.on('load.channel', this.autoClaimBonusPoints);
```
Comment (night) [inline src/modules/channel_points/index.js:29]: we don't need to load this twice, as dom listeners persist

File: src/modules/channel_points/index.js [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+
+const CLAIM_BUTTON_SELECTOR = '.claimable-bonus__icon';
+
+let channelPointsListener;
+
+class ChannelPoints {
+    constructor() {
+        settings.add({
+            id: 'autoClaimBonusPoints',
+            name: 'Auto-Claim Bonus Points',
```
Comment (night) [inline src/modules/channel_points/index.js:14]: `Auto-Claim Channel Points`

File: src/modules/channel_points/index.js [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+const domObserver = require('../../observers/dom');
+
+const CLAIM_BUTTON_SELECTOR = '.claimable-bonus__icon';
+
+let channelPointsListener;
+
+class ChannelPoints {
+    constructor() {
+        settings.add({
+            id: 'autoClaimBonusPoints',
```
Comment (night) [inline src/modules/channel_points/index.js:13]: `autoClaimBonusChannelPoints`

File: src/modules/channel_points/index.js [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+            id: 'hideChannelPoints',
+            name: 'Hide Channel Points',
+            defaultValue: false,
+            description: 'Hides channel points from the chat UI to reduce clutter',
+        });
+
+        settings.on('changed.hideChannelPoints', this.hideChannelPoints);
+        settings.on('changed.autoClaimBonusPoints', this.autoClaimBonusPoints);
+
+        this.autoClaimBonusPoints();
```
Comment (night) [inline src/modules/channel_points/index.js:28]: we are never loading the `hideChannelPoints`

**Review by night — CHANGES_REQUESTED** [review]
